### PR TITLE
Egyptian Food World website

### PR DESCRIPTION
**Pull request by Maha-Magdy**
We have managed to complete the following: 
1- Setup the project environment
2- Design the header and the footer
3- Structure and style the website homepage and the comments popup.
4- Fetch the meals listed from the API using the filter to get only the Egyptian meals.
5- Make a counter at the header with the number of items on the homepage.
6- Add functionality to add a like and the likes counter is updated once you like.
7- Show more details about the item when clicking on comment.
8- Create add a comment section to be able to add a comment.
9- Post the added comment to the involvement API and get it back to show the current comments an item has.
10- Add a comments counter to show the number of comments.
11- Make tests for the likes counter and the comments counter.

### CODE REVIEW

**Review by vichuge — CHANGES_REQUESTED** [review]
# Project invalidated 🚫 

Hi there, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

cat_massage

## Why does this happen ❓ ❓ ❓ 

You have submitted a project, that has some linters errors, please fix them before to ask another code review. 
image


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/homepage.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/* eslint-disable import/prefer-default-export */
+/* eslint-disable no-restricted-syntax */
+
+import Meal from './meal.js';
+
+export function updateMealsNumber(mealsList) {
```
Comment (vichuge) [inline src/homepage.js]: - [x] It looks like here we could use ES6 best practices (in this case arrow function). 🐱

File: src/homepage.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  document.getElementById('meals-num').textContent = mealsList.length;
+}
+
+function createMealsObject(mealsList, likesList) {
```
Comment (vichuge) [inline src/homepage.js]: - [x] It looks like here we could use ES6 best practices (in this case arrow function). 🐱

File: src/homepage.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      meal.strMeal,
+      meal.strMealThumb,
+      meal.idMeal,
+      likes,
+    );
+    mealObj.drawTheMeal();
+  }
+}
+
+async function egyptianMeals() {
```
Comment (vichuge) [inline src/homepage.js]: - [x] It looks like here we could use ES6 best practices (in this case arrow function). 🐱

File: src/meal.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable radix */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable array-callback-return */
+
+async function listComments(itemId) {
```
Comment (vichuge) [inline src/meal.js]: - [x] It looks like here we could use ES6 best practices (in this case arrow function). 🐱

File: src/meal.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+  title.innerHTML = `<h5>Comments</h5><h5 id="counter">(${comments.length})</h5>`;
+  comments.map((comment) => {
+    ul.innerHTML += `<p style="width: 100%; display: flex; justify-content: space-between"><span><strong>${comment.username}:</strong> ${comment.comment}</span> <span>${comment.creation_date}</span</p> <br>`;
+  });
+
+  commentsList.append(title);
+  commentsList.append(ul);
+}
+
+async function addComment(itemId) {
```
Comment (vichuge) [inline src/meal.js]: - [x] It looks like here we could use ES6 best practices (in this case arrow function). 🐱

File: src/meal.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable radix */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable array-callback-return */
```
Comment (vichuge) [inline src/meal.js]: - [x] It looks like you're using some disablers here, it's ok if we're struggling with some issues and then we use it to continue 1 or 2 times, please, try to do the code without disabling a lot of functions here. 🌵

File: README.md [hunk 7]
```diff
@@ -1 +1,71 @@
...
+## Built With
+
+- HTML, CSS, JavaScript;
+- Jest, & Webpack;
+- VisualStudio Code, Git, & GitHub;
+
+## Live Demo
+
+[Live Demo Link]( https://maha-magdy.github.io/egyptian-food-world/ )
+
```
Comment (vichuge) [inline README.md:23]: - [x] I'm afraid I couldn't find the video presentation on the pull request description or the readme file, please add it. 🐶

**Review by vichuge — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi dear student, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 
You used proper GitHub Flow ✔️
Your linter checks are all passing ✔️
Your PR messages are descriptive ✔️

## Aspects to improve ♻️

- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by descholar-ceo — APPROVED** [review]
Hi @Maha-Magdy and @MaiRizk , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**